shared/domain/src/main/java/com/ivy/domain/usecase/BalanceBuilder.kt

```diff
@@ -0,0 +1,63 @@
+package com.ivy.domain.usecase
+
+import com.ivy.data.model.primitive.AssetCode
+import com.ivy.data.model.primitive.NonZeroDouble
+import com.ivy.data.model.primitive.PositiveDouble
+
+class BalanceBuilder {
```

Comment:
You can start with unit tests for this class. See the ones that I did for the other StatSummaryBuilder

Consider adding:
- unit tests
- property based tests verifying Math properties that you can identify (e.g. the order of transactions doesn't matter for the balance)

Reply:
Ok. I'll read about `PositiveDouble` to solve that Left case and then move on to the tests